### DISCUSSION TITLE
Accessors in GraphQL Resources

### DISCUSSION THREAD

**Discussion by ThisaruGuruge** [ballerina-platform/ballerina-standard-library#757] [General]
Currently, the Ballerina GraphQL module allows all the types of accessors for a resource function.

I think we should only support the `get` accessor for GraphQL queries and the `post` accessor for mutations (when we support mutations). 

@shafreenAnfar WDYT?

**Comment by shafreenAnfar** [comment]
Well, I need to refresh my GraphQL theory. So @ThisaruGuruge correct me if I am wrong. 

AFAIR, GraphQL specification does not mandate sending `query` operations as `get` or `post`. In fact, I believe most of the `query` operations would be sent as `post` request. I think we shouldn't mix HTTP or REST concepts with GraphQL concepts.  

Anyway, now that you brought this up, I am thinking if we should have three accessors as `query`, `mutation` and `subscription` for GraphQL.

Also, when designing APIs, I think it is always good to follow the below principle. So that we can serve both good citizens as well as bad citizens of a given community. Because usually in large communities, there could be slightly different implementations of the same spec. HTTP is notoriously famous for this. 

"Be conservative in what you do, be liberal in what you accept from others"

**Comment by ThisaruGuruge** [comment]
Great explanation! I agree. Let's go with the `query` and `mutation` accessors for GraphQL.

**Reply by shafreenAnfar** [reply]
I think we need to consult James opinion on this also before we proceed. I remember he was thinking of using remote methods for mutations.
